### PR TITLE
don't emit log if no changes occur

### PR DESCRIPTION
**Pull request by se3000**

### CODE REVIEW

File: core/internal/features_test.go [hunk 1]
```diff
@@ -285,7 +285,8 @@ func TestIntegration_EthLog(t *testing.T) {
 	assert.Equal(t, address, initr.Address)
 
 	logs <- cltest.LogFromFixture(t, "testdata/requestLog0original.json")
-	cltest.WaitForRuns(t, j, app.Store, 1)
+	jrs := cltest.WaitForRuns(t, j, app.Store, 1)
+	cltest.WaitForJobRunToComplete(t, app.Store, jrs[0])
```
Comment (samsondav) [inline core/internal/features_test.go:289]: This seems unrelated, is it to fix a flakey test?

Comment (se3000) [reply core/internal/features_test.go:289]: Yeah, popped up on a CI build so I included it.
